packages/datadog-instrumentations/test/helpers/promise.js

```diff
@@ -13,54 +13,59 @@ module.exports = (name, factory, versionRange) => {
         if (versionRange && !semver.intersects(version, versionRange)) return
 
         beforeEach(() => {
-          const prq = proxyquire.noPreserveCache()
-          const moduleExports = prq(`../../../../versions/${name}@${version}`, {}).get()
+          const moduleExports = require(`../../../../versions/${name}@${version}`, {}).get()
 
           Promise = factory ? factory(moduleExports) : moduleExports
         })
 
         it('should run the then() callbacks in the context where then() was called', () => {
-          const store = storage.getStore()
+          const store = 'store'
 
           let promise = new Promise((resolve, reject) => {
             setImmediate(() => {
-              storage.run(store, () => {
+              storage.run('promise', () => {
```

Comment:
If this is here, do we still need the surrounding setImmediate?

Reply:
I think it's better to make it asynchronous yes. There have been issues in the past where the scope would leak synchronously, which can result in unexpected results. By making it asynchronous, it would leak in a different scope.